[emit] Implement emit functions for memory access ops with four variants

dongju-chae opened the pull request:
This PR implements emit functions for memory access ops (str/ldr) with four variants:
- literal (label), unsigned offset, signed offset (pre-/post-index) and extended register.

Signed-off-by: Dongju Chae <dongju.chae@samsung.com>

myungjoo commented on orc/orcarm.c:
```diff
@@ -417,34 +417,14 @@ orc_arm_emit_load_reg (OrcCompiler *compiler, int dest, int src1, int offset)
+
+  ORC_ASM_CODE(compiler,"  ldr %s, [%s, #%d]\n",
+      orc_arm_reg_name (dest),
+      orc_arm_reg_name (src1), offset);
```
Are you reverting 64bit code of load reg?

dongju-chae replied:
No, I just want to move this impl. to more generic one, `orc_arm64_emit_mem`, because load/store instructions have many variants.
This function, `orc_arm_emit_load_reg`, will be used only for arm32.

myungjoo commented on orc/orcarm.c:
```diff
@@ -1016,3 +998,189 @@ orc_arm64_emit_am (OrcCompiler *p, OrcArm64RegBits bits, OrcArm64DP opcode,
+        }
+
+        if (imm != 0) {
+          snprintf (opt_rn, ARM64_MAX_OP_LEN - 1, ", [%s", orc_arm64_reg_name(Rn, bits));
```
I thought the size of snprintf does not count the terminating null byte; however, the manual says:
```
       The functions snprintf() and vsnprintf() write at most size bytes (including the terminating null byte ('\0')) to str.
```

myungjoo reviewed (approved):
LGTM

Commit:
[emit] Implement emit functions for memory access ops with variants

This commit implements emit functions for memory access ops (str/ldr) with four variants:
- literal, unsigned offset, signed offset (pre-/post-index) and extended register.

Signed-off-by: Dongju Chae <dongju.chae@samsung.com>
